src/main/java/games/strategy/engine/framework/ProcessRunnerUtil.java

```diff
@@ -29,21 +30,17 @@ public static void populateBasicJavaArgs(final List<String> commands) {
     populateBasicJavaArgs(commands, System.getProperty("java.class.path"));
   }
 
-  public static void populateBasicJavaArgs(final List<String> commands, final long maxMemory) {
+  public static void populateBasicJavaArgs(final List<String> commands, final Optional<String> maxMemory) {
```

Comment:
This overload is only called in one place (`MapCreator#runUtility()`) with a non-empty `Optional` value for `maxMemory`.  You can probably change the type of `maxMemory` back to `long` and do the `Optional` encapsulation here.  (Note that it appears `maxMemory` is measured in B, not MiB; might want to make that clear in the parameter name.)

```diff
diff --git a/src/main/java/games/strategy/engine/framework/ProcessRunnerUtil.java b/src/main/java/games/strategy/engine/framework/ProcessRunnerUtil.java
index 1cba740..635b4ad 100644
--- a/src/main/java/games/strategy/engine/framework/ProcessRunnerUtil.java
+++ b/src/main/java/games/strategy/engine/framework/ProcessRunnerUtil.java
@@ -30,8 +30,9 @@ public class ProcessRunnerUtil {
     populateBasicJavaArgs(commands, System.getProperty("java.class.path"));
   }
 
-  public static void populateBasicJavaArgs(final List<String> commands, final Optional<String> maxMemory) {
-    populateBasicJavaArgs(commands, System.getProperty("java.class.path"), maxMemory);
+  public static void populateBasicJavaArgs(final List<String> commands, final long maxMemoryInBytes) {
+    populateBasicJavaArgs(commands, System.getProperty("java.class.path"),
+        Optional.of((maxMemoryInBytes / 1024L / 1024L) + "M"));
   }
 
   static void populateBasicJavaArgs(final List<String> commands, final String newClasspath) {
diff --git a/src/main/java/tools/map/making/MapCreator.java b/src/main/java/tools/map/making/MapCreator.java
index 24dcb75..127093c 100644
--- a/src/main/java/tools/map/making/MapCreator.java
+++ b/src/main/java/tools/map/making/MapCreator.java
@@ -10,7 +10,6 @@ import java.awt.event.FocusListener;
 import java.io.File;
 import java.util.ArrayList;
 import java.util.List;
-import java.util.Optional;
 
 import javax.swing.Action;
 import javax.swing.Box;
@@ -524,7 +523,7 @@ public class MapCreator extends JFrame {
 
   private static void runUtility(final Class<?> javaClass) {
     final List<String> commands = new ArrayList<>();
-    ProcessRunnerUtil.populateBasicJavaArgs(commands, Optional.of(s_memory + "M"));
+    ProcessRunnerUtil.populateBasicJavaArgs(commands, s_memory);
     if (s_mapFolderLocation != null && s_mapFolderLocation.exists()) {
       // no need for quotes, that will just screw up the process builder
       commands.add("-D" + TRIPLEA_MAP_FOLDER + "=" + s_mapFolderLocation.getAbsolutePath());
```